langlib/lang.query/src/main/ballerina/src/lang.query/types.bal

```diff
@@ -66,9 +66,16 @@ class _StreamPipeline {
         self.resType = resType;
     }
 
-    public function next() returns _Frame|error? {
+    public isolated function next() returns _Frame|error? {
         _StreamFunction sf = self.streamFunction;
-        return sf.process();
+        var res = internal:invokeAsExternal(sf.process);
+        if (res is _Frame) {
+            return res;
+        } else if (res is error) {
+            return res;
```

Comment:
Can't we do
```suggestion
        if (res is _Frame|error) {
            return res;
```

Reply:
fixed in: d97ca9cdac22409c90315440b107d70588065625